### ISSUE TITLE
ASP .NET core async requests - performance

### ISSUE THREAD

**Issue by marian-margeta** [aspnet/Home#2848]
I created simple ASP .net core 2.0 web-api application to try performance of async requests. In the `SampleController` I created `Get()` method where I call 10s async `Delay` and then I return the time when the request was called.  However, if I run the server (in console) and manually call this web method from my browser in multiple tabs approximately simultaneously, the first request ends after ~10s and the rest after ~15s. Why there is such big delay? 

I tried to compile as Debug or Release but I got the same result.

**SampleController.cs**
```csharp
[Route("/")]
public class SampleController : Controller
{
    [HttpGet]
    public async Task<string> Get()
    {
        var time = DateTime.Now;

        await Task.Delay(TimeSpan.FromSeconds(10));

        return time.ToString();
    }
}
```

**Program.cs**
```csharp
public class Program
{
    public static void Main(string[] args)
    {
        BuildWebHost(args).Run();
    }

    public static IWebHost BuildWebHost(string[] args) =>
        WebHost.CreateDefaultBuilder(args)
            .UseStartup<Startup>()
            .Build();
}
```

**Startup.cs**
```csharp
public class Startup
{
    public Startup(IConfiguration configuration)
    {
        Configuration = configuration;
    }

    public IConfiguration Configuration { get; }

    // This method gets called by the runtime. Use this method to add services to the container.
    public void ConfigureServices(IServiceCollection services)
    {
        services.AddMvc();
    }

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    public void Configure(IApplicationBuilder app, IHostingEnvironment env)
    {
        app.UseMvc();
    }
}
```

**Comment by davidfowl** [comment]
Try a different client?

**Comment by marian-margeta** [comment]
What client do you mean?

**Comment by davidfowl** [comment]
Not a browser or a different browser, or fiddler, or wrk, any client really. The browser does weird things sometimes. Also, requests may not exactly end after 10 seconds  depending on the load, continuations for timer callbacks are scheduled on the thread pool and if that's overloaded then the timer won't fire on time. Though a 5 second difference isn't likely if there isn't much load on the system. 

How are you measuring time response time?

**Comment by marian-margeta** [comment]
I measured the duration of the requests according to the console logs. By default, each request prints info about its duration, e.g.:
```
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
Request finished in 10011.35ms 200 text/plain; charset=utf-8
```
But in fact, each request has approximately 10s duration. The problem is, when the request started. The first started immediately and the rest after ~5s. This was tested with Firefox 58.0. 

And you are right, the problem is in the client. I tested it with chrome, and it was even worse. Each request starts when the previous one ends, so after 10s. When I used postman, each request started immediately. 

So now it makes sense, maybe the browser is trying to do some optimization. Thanks @davidfowl
